scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,272 @@
+---
+title: "Setup: Introduction Module"
+author: Stephanie Spielman for the Data Lab
+date: "`r Sys.Date()`"
+output: 
+  html_notebook:
+    toc: true
+    toc_depth: 3
+    toc_float: true
+    number_sections: true
+---
+
+```{r setup, include=FALSE}
+set.seed(2022)
+
+library(SingleCellExperiment)
+library(magrittr)
+library(SingleR) # annotation
+library(bluster) # clustering to give the people their UMAP colors
+```
+
+To get the raw data:
+```markdown
+NAME=10k_PBMCs_TotalSeq_B_3p_raw_feature_bc_matrix.tar.gz
+wget https://cf.10xgenomics.com/samples/cell-exp/6.0.0/10k_PBMCs_TotalSeq_B_3p/$NAME
+tar -xf $NAME
+# This gets unpacked into `raw_feature_bc_matrix`
+
+
+# For posterity, first attempt was with MALT, but it had wildly low ADT counts
+#NAME=malt_10k_protein_v3_raw_feature_bc_matrix.tar.gz
+#wget https://cf.10xgenomics.com/samples/cell-exp/3.0.0/malt_10k_protein_v3/$NAME
+```
+
+## Filtering and normalization
+
+```{r}
+
+pbmc_10x_dir <- "raw_feature_bc_matrix"
+# for now..
+mito_file <- "https://raw.githubusercontent.com/AlexsLemonade/scpca-downstream-analyses/main/reference-files/Homo_sapiens.GRCh38.104.mitogenes.txt"
+
+
+
+pbmc_sce_raw <- DropletUtils::read10xCounts(pbmc_10x_dir)
+mito_genes <- readr::read_tsv(mito_file, 
+                              col_names=FALSE) %>%
+  # only keep relevant mitochondrial genes
+  dplyr::filter(X1 %in% rownames(pbmc_sce_raw)) %>%
+  dplyr::pull(X1)
+```
+
+
+### Initial Mitochondria QC:
+
+Note that the `addPer<X>QC()` functions should be run before we split out the ADTs into their own altExp so that the appropriate QC columns actually get ported into the altExp.
+
+```{r}
+pbmc_sce_percellqc <- scater::addPerCellQC(
+  pbmc_sce_raw, 
+  subsets = list(mito = mito_genes))
+
+# we'll need this later, so pop it in now
+pbmc_sce_percellqc <- scater::addPerFeatureQC(pbmc_sce_percellqc)
```

Comment:
This should happen after filtering, as values change with filtering.

Reply:
Ah, I realize I meant to delete this since it is not strictly needed for ADT normalization for this data set (it was when I was exploring the MALT data).